README.md
# tt_news

TYPO3 Extension - News

Website news with front page teasers and article handling inside. 

AKA: **"the generic record handler"**

## TYPO3 Extension Repository
 
[https://typo3.org/extensions/repository/view/tt_news](https://typo3.org/extensions/repository/view/tt_news)



## Composer

[https://packagist.org/packages/rupertgermann/tt_news](https://packagist.org/packages/rupertgermann/tt_news)

    composer require rupertgermann/tt_news
    
## Compatibility

branch master requires at least TYPO3 8.7.0 and is compatible with TYPO3 8.7 LTS up to TYPO3 9.5 LTS.

branch feature/tt-news-76 works with TYPO3 7.6 LTS

For tt_news for older TYPO3 versions take a look at [https://extensions.typo3.org/extension/tt_news/](https://extensions.typo3.org/extension/tt_news/)  and scroll down to "Version History".      

ChangeLog
2019-02-12  netz.coop eG <kontakt@netz.coop>
	* https://github.com/rupertgermann/tt_news/issues/119 BUGFIX 2: SQL error: 'Incorrect integer value: '' for column 'shortcut' at row 1' (tt_news_cat:NEW...)

2012-05-22  Oliver Hader <oliver.hader@typo3.org>
	* Raised version to 3.3.0
	* Added compatibility layer for removed methods in TYPO3 6.0

2012-04-18  Rupert Germann <rupi@gmx.li>
	* release of tt_news 3.2.1 in TER
	* fixed bug 18507: using "cacheTable" is deprecated and broken in core master

2012-03-15  Rupert Germann <rupi@gmx.li>
	* release of tt_news 3.2 in TER
	
2012-02-17 Oliver Hader <oliver.hader@typo3.org>
	* !!! Fixed bug #12626: No workspace preview for translated news (order changed to version overlay, localization overlay)

2012-02-15 Oliver Hader <oliver.hader@typo3.org>
	* Raise required TYPO3 version to 4.3.0
	* Fixed bug #9234: Parant localization record cannot be modified (thanks to Robert Heel)
	* Fixed bug #14187: Translated news not shown if news in parent record of type internal/external link (thanks to Robert Heel)

2011-12-30 Rupert Germann <rupi@gmx.li>
	* fixed bug #18392: Typo in mod1/index.php (thanks to Andrey Aksenov)

2011-12-18 Oliver Hader <oliver.hader@typo3.org>
	* Fixed bug #18369: t3lib_htmlmail removed in TYPO3 4.7, tt_news crashes

2011-11-06 Oliver Hader <oliver.hader@typo3.org>
	* Fixed bug #18341: LocalLang labels are not correctly overridden in TYPO3 4.6

2011-10-16 Rupert Germann <rupi@gmx.li>
	* fixed bug #18268: No news records since TYPO3 4.6beta2

2011-04-06 Rupert Germann <rupi@gmx.li>
	* bugfix: catOrderBy was overwritten in a wrong way when displaying related news by category causing an SQL error
	* fixed bug #13330: error: Invalid database result resource detected (Thanks to Philipp Gampe)
	* fixed bug #13232: check for category access does not work, if no restriction is applied (Thanks to Andreas Kiessling)
	* fixed bug #16351: Missing $pi_checkCHash (Thanks to Robert Heel)

2011-03-11 Rupert Germann <rupi@gmx.li>
	* bugfix: the direct link in EM that points to to the updater script didn't work in TYPO3 4.5

2011-03-04 Rupert Germann <rupi@gmx.li>
	* bugfix: "relatedNewsByCategory" and "renderRelatedNewsAsList" didn't work together
	* fixed bug #17870: Table tt_news_cache_tags is not flushed (thanks to Daniel Minder)
	* fixed bug #16715: Caching framework configuration from localconf.php was overwritten by ext_localconf (thanks to Ralf Strobel)

2011-03-04 Rupert Germann <rupi@gmx.li>
	* fixed bug #15322: remove folder icon code which is deprecated since TYPO3 v4.4 (thanks to Simon Schick)
	* fixed bug #17458: Hostname appended twice to internal links in feed when config.absRefPrefix set (thanks to Christian Weiske)
	* fixed bug #14450: local_cObj is overwritten with data from related news by category (thanks to Sven Burkert)

2011-03-03 Rupert Germann <rupi@gmx.li>
	* fixed bug #16876: Plugin-CE not editable / t3lib_error_Exception / t3lib_iconWorks::getSpriteIconForRecord() (thanks to Thomas Oliver Moll)
	* fixed bug #14001 (part 2): Deleted flag was not taken into account when checking categories for localized records
	* fixed bug #13752: tt_news crash in single view (PHP Error: Argument 2 passed to t3lib_div::array_merge_recursive_overrule() 
	  must be an array, null given) (Thanks to Geoffroy Cochard)

2011-03-02 Rupert Germann <rupi@gmx.li>
	* fixed bug #14001: User can't edit news with deleted categories

2011-02-01 Rupert Germann <rupi@gmx.li>
	* new feature: it's now possible to render related news with an html template.
	* new feature: by setting "restrictListToThisTypes" it's possible to render a list view from news which
	  shows only news having the selected type.
	* raised compatibility version to TYPO3 4.5


2009-12-07 Rupert Germann <rupi@gmx.li>
	* fixed bug #13753: Category menu in mode "nested wraps" is broken. Thanks to Jochen Weiland for discovering 
	this issue and to Ingo Renner for sponsorimg this bug at the T3Board10 bug auction.
	
2009-12-07 Rupert Germann <rupi@gmx.li>
	* fixed bug #12944 (follow-up to #12848):  !!! Database change !!! unknown db field 'tags' in table tt_news_cache (thanks to Sven Juergens)

2009-12-02 Rupert Germann <rupi@gmx.li>
	* fixed bug #12848: !!! Database change !!! using caching framework for internal caching fails with exception. 
	* fixed bug #12808: Duplicated '?>' in class.tx_ttnews_cache.php (thanks to Stephan Bauer)
	* release of tt_news 3.0.1

2009-11-27 Rupert Germann <rupi@gmx.li>
	* fixed bug #9215: tt_news 3.0 category ajaxtree looses language parameter on ajax expand/collapse 
	* fixed bug #12757: IFNULL statements in tt_news queries are not DBAL compatible 
	* updated context sensitive help (CSH)
	* added missing xclass statement to class tt_news_cache
	* updated ext_emconf.php	
	* release of tt_news 3.0
	
2009-11-26  Rupert Germann <rupi@gmx.li>
	* added flashmessages icons for older TYPO3 versions to folder compat/gfx
	* changed "disallowed categories" message in tceforms to use flashmessages 
	* changed tt_news form: moved "hidden" field to the "General" tab and moved author-email to a palette (Thanks to Georg Ringer)
	* follow-up to feature #12118: removed tab "Typoscript" from flexform and moved the new "typoscript" field to the "other settings" tab
	* follow-up to feature #12118: added a condtion to the flexform field "typoscript" which hides this field from non-admin users

tca.php
<?php
/**
 * $Id$
 */

	// get extension confArr
$confArr = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['tt_news']);
	// switch the use of the "StoragePid"(general record Storage Page) for tt_news categories
$fTableWhere = ($confArr['useStoragePid'] ? 'AND tt_news_cat.pid=###STORAGE_PID### ' : '');
	// page where records will be stored in that have been created with a wizard
$sPid = ($fTableWhere ? '###STORAGE_PID###' : '###CURRENT_PID###');
	// l10n_mode for text fields
$l10n_mode = ($confArr['l10n_mode_prefixLangTitle'] ? 'prefixLangTitle' : '');
$l10n_mode_author = ($confArr['l10n_mode_prefixLangTitle'] ? 'mergeIfNotBlank' : '');
	// l10n_mode for the image field
$l10n_mode_image = ($confArr['l10n_mode_imageExclude'] ? 'exclude' : 'mergeIfNotBlank');
	// hide new localizations
$hideNewLocalizations = ($confArr['hideNewLocalizations'] ? 'mergeIfNotBlank' : '');
// ******************************************************************
// This is the standard TypoScript news table, tt_news
// ******************************************************************
$TCA['tt_news'] = Array (
	'ctrl' => $TCA['tt_news']['ctrl'],
	'interface' => Array (
		'showRecordFieldList' => 'title,hidden,datetime,starttime,archivedate,category,author,author_email,short,image,imagecaption,links,related,news_files'
	),
	'columns' => Array (
		'starttime' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.starttime',
			'config' => Array (
				'type' => 'input',
				'size' => '10',
				'max' => '20',
				'eval' => 'datetime',
				'checkbox' => '0',
				'default' => '0'
			)
		),
		'endtime' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.endtime',
			'config' => Array (
				'type' => 'input',
				'size' => '8',
				'max' => '20',
				'eval' => 'datetime',
				'checkbox' => '0',
				'default' => '0',
				'range' => Array (
					'upper' => mktime(0,0,0,12,31,2020),
					'lower' => mktime(0,0,0,date('m')-1,date('d'),date('Y'))
				)
			)
		),
		'hidden' => Array (
			'l10n_mode' => $hideNewLocalizations,
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.hidden',
			'config' => Array (
				'type' => 'check',
				'default' => '1'
			)
		),
		'fe_group' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.fe_group',
			'config' => Array (
				'type' => 'select',
				'size' => 5,
				'maxitems' => 20,
				'items' => Array (
					Array('LLL:EXT:lang/locallang_general.php:LGL.hide_at_login', -1),
					Array('LLL:EXT:lang/locallang_general.php:LGL.any_login', -2),
					Array('LLL:EXT:lang/locallang_general.php:LGL.usergroups', '--div--')
				),
				'exclusiveKeys' => '-1,-2',
				'foreign_table' => 'fe_groups'
			)
		),
 		'title' => Array (
 			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.title',
			'l10n_mode' => $l10n_mode,
 			'config' => Array (
 				'type' => 'input',
 				'size' => '40',
 				'max' => '256'
 			)
 		),
		'ext_url' => Array (
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.external',
			'config' => Array (
				'type' => 'input',
				'size' => '40',
				'max' => '256',
				'wizards' => Array(
					'_PADDING' => 2,
					'link' => Array(
						'type' => 'popup',
						'title' => 'Link',
						'icon' => 'link_popup.gif',
						'script' => 'browse_links.php?mode=wizard',
						'JSopenParams' => 'height=300,width=500,status=0,menubar=0,scrollbars=1'
					)
				)
			)
		),
		'bodytext' => Array (
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.text',
			'l10n_mode' => $l10n_mode,
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '5',
				'softref' => 'typolink_tag,images,email[subst],url',
				'wizards' => Array(
					'_PADDING' => 4,
					'RTE' => Array(
						'notNewRecords' => 1,
						'RTEonly' => 1,
						'type' => 'script',
						'title' => 'LLL:EXT:cms/locallang_ttc.php:bodytext.W.RTE',
						'icon' => 'wizard_rte2.gif',
						'script' => 'wizard_rte.php',
					),
				)
			)
		),
		'no_auto_pb' => Array (
			'l10n_mode' => 'mergeIfNotBlank',
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.no_auto_pb',
			'config' => Array (
				'type' => 'check'
			)
		),
		'short' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.subheader',
			'l10n_mode' => $l10n_mode,
			'config' => Array (
				'type' => 'text',
				'cols' => '40',
				'rows' => '3'
			)
		),
		'type' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.type',
			'config' => Array (
				'type' => 'select',
				'items' => Array (
					Array('LLL:EXT:tt_news/locallang_tca.xml:tt_news.type.I.0', 0),
					Array('LLL:EXT:tt_news/locallang_tca.xml:tt_news.type.I.1', 1),
					Array('LLL:EXT:tt_news/locallang_tca.xml:tt_news.type.I.2', 2)
				),
				'default' => 0
			)
		),
		'datetime' => Array (
			'l10n_mode' => 'mergeIfNotBlank',
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.datetime',
			'config' => Array (
				'type' => 'input',
				'size' => '10',
				'max' => '20',
				'eval' => 'datetime',
				'default' => mktime(date("H"),date("i"),0,date("m"),date("d"),date("Y"))
				)
		),
		'archivedate' => Array (
			'l10n_mode' => 'mergeIfNotBlank',
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.archivedate',
			'config' => Array (
				'type' => 'input',
				'size' => '10',
				'max' => '20',
				'eval' => 'date',
				'default' => '0'
			)
		),
		'image' => Array (
			'exclude' => 1,
			'l10n_mode' => $l10n_mode_image,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.images',
			'config' => Array (
				'type' => 'group',
				'internal_type' => 'file',
				'allowed' => $GLOBALS['TYPO3_CONF_VARS']['GFX']['imagefile_ext'],
				'max_size' => '10000',
				'uploadfolder' => 'uploads/pics',
				'show_thumbs' => '1',
				'size' => 3,
				'autoSizeMax' => 15,
				'maxitems' => '99',
				'minitems' => '0'
			)
		),
		'imagecaption' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.caption',
			'l10n_mode' => $l10n_mode,
			'config' => Array (
				'type' => 'text',
				'cols' => '30',
				'rows' => '3'
			)
		),
		'imagealttext' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.imagealttext',
			'l10n_mode' => $l10n_mode,
			'config' => Array (
				'type' => 'text',
				'cols' => '20',
				'rows' => '3'
			)
		),
		'imagetitletext' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.imagetitletext',
			'l10n_mode' => $l10n_mode,
			'config' => Array (
				'type' => 'text',
				'cols' => '20',
				'rows' => '3'
			)
		),
		'author' => Array (
			'exclude' => 1,
			'l10n_mode' => $l10n_mode_author,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.author',
			'config' => Array (
				'type' => 'input',
				'size' => '20',
				'eval' => 'trim',
				'max' => '80'
			)
		),
		'author_email' => Array (
			'exclude' => 1,
			'l10n_mode' => $l10n_mode_author,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.email',
			'config' => Array (
				'type' => 'input',
				'size' => '20',
				'eval' => 'trim',
				'max' => '80'
			)
		),
		'related' => Array (
			'exclude' => 1,
			'l10n_mode' => 'exclude',
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.related',
			'config' => Array (
				'type' => 'group',
				'internal_type' => 'db',
					'allowed' => 'tt_news,pages',
					'MM' => 'tt_news_related_mm',
				'size' => '3',
				'autoSizeMax' => 10,
				'maxitems' => '200',
				'minitems' => '0',
				'show_thumbs' => '1',
				'wizards' => array(
					'suggest' => array(
						'type' => 'suggest'
					)
				)
			)
		),
		'keywords' => Array (
			'l10n_mode' => 'mergeIfNotBlank',
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.keywords',
			'config' => Array (
				'type' => 'text',
				'cols' => '40',
				'rows' => '3'
			)
		),
		'links' => Array (
			'l10n_mode' => $l10n_mode_author,
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.links',
			'config' => Array (
				'type' => 'text',
				'cols' => '40',
				'rows' => '3'
			)
		),
		'category' => Array (
			'exclude' => 1,
		#	'l10n_mode' => 'exclude', // the localizalion mode will be handled by the userfunction
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.category',
			'config' => Array (
				'type' => 'select',
				'form_type' => 'user',
				'userFunc' => 'tx_ttnews_TCAform_selectTree->renderCategoryFields',
				'treeView' => 1,
				'foreign_table' => 'tt_news_cat',
				'autoSizeMax' => 50,
				'minitems' => $confArr['requireCategories'] ? 1 : 0,
				'maxitems' => 500,
				'MM' => 'tt_news_cat_mm',

			)
		),
		'page' => Array (
			'exclude' => 1,
			'l10n_mode' => 'exclude',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.shortcut_page',
			'config' => Array (
				'type' => 'group',
				'internal_type' => 'db',
				'allowed' => 'pages',
				'size' => '1',
				'maxitems' => '1',
				'minitems' => '0',
				'show_thumbs' => '1'
			)
		),
		'news_files' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:cms/locallang_ttc.php:media',
			'config' => Array (
				'type' => 'group',
				'internal_type' => 'file',
				'allowed' => '',	// Must be empty for disallowed to work.
				'disallowed' => 'php,php3',
				'max_size' => '10000',
				'uploadfolder' => 'uploads/media',
				'show_thumbs' => '1',
				'size' => '3',
				'autoSizeMax' => '10',
				'maxitems' => '100',
				'minitems' => '0'
			)
		),
		'sys_language_uid' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.language',
			'config' => Array (
				'type' => 'select',
				'foreign_table' => 'sys_language',
				'foreign_table_where' => 'ORDER BY sys_language.title',
				'items' => Array(
					Array('LLL:EXT:lang/locallang_general.php:LGL.allLanguages',-1),
					Array('LLL:EXT:lang/locallang_general.php:LGL.default_value',0)
				)
			)
		),
		'l18n_parent' => Array (
			'displayCond' => 'FIELD:sys_language_uid:>:0',
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.l18n_parent',
			'config' => Array (
				'type' => 'select',
				'items' => Array (
					Array('', 0),
				),
				'foreign_table' => 'tt_news',
				'foreign_table_where' => 'AND tt_news.pid=###CURRENT_PID### AND tt_news.sys_language_uid IN (-1,0)',
			)
		),
		'l18n_diffsource' => Array(
			'config'=>array(
				'type'=>'passthrough')
		),
		't3ver_label' => Array (
			'displayCond' => 'FIELD:t3ver_label:REQ:true',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.versionLabel',
			'config' => Array (
				'type'=>'none',
				'cols' => 27
			)
		),

		'editlock' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_tca.xml:editlock',
			'config' => Array (
				'type' => 'check'
			)
		),


		/**
		 * The following fields have to be configured here to get them processed by the listview in the tt_news BE module
		 * they should never appear in the 'showitem' list as editable fields, though.
		 */
		'uid' => Array (
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.uid',
			'config' => Array (
				'type' => 'none'
			)
		),
		'pid' => Array (
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.pid',
			'config' => Array (
				'type' => 'none'
			)
		),
		'tstamp' => Array (
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news.tstamp',
			'config' => Array (
				'type' => 'input',
				'eval' => 'datetime',
			)
		),

	),
	'types' => Array (
		'0' => Array('showitem' =>
			'hidden, type;;;;1-1-1,title;;;;2-2-2,short,bodytext;;2;richtext:rte_transform[flag=rte_enabled|mode=ts];4-4-4,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.special, datetime;;;;2-2-2,archivedate,author;;3;; ;;;;2-2-2,
				keywords;;;;2-2-2,sys_language_uid;;1;;3-3-3,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.media, image;;;;1-1-1,imagecaption;;5;;,links;;;;2-2-2,news_files;;;;4-4-4,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.catAndRels, category;;;;3-3-3,related;;;;3-3-3,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.access, starttime,endtime,fe_group,editlock,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.extended,
			'),

		'1' => Array('showitem' =>
			'hidden, type;;;;1-1-1,title;;;;2-2-2,page,short,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.special, datetime;;;;2-2-2,archivedate,author;;3;; ;;;;2-2-2,
				keywords;;;;2-2-2,sys_language_uid;;1;;3-3-3,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.media, image;;;;1-1-1,imagecaption;;5;;,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.categories, category;;;;3-3-3,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.access, starttime,endtime,fe_group,editlock,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.extended,
			'),

		'2' => Array('showitem' =>
			'hidden, type;;;;1-1-1,title;;;;2-2-2,ext_url,short,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.special, datetime;;;;2-2-2,archivedate,author;;3;; ;;;;2-2-2,
				keywords;;;;2-2-2,sys_language_uid;;1;;3-3-3,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.media, image;;;;1-1-1,imagecaption;;5;;,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.categories, category;;;;3-3-3,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.access, starttime,endtime,fe_group,editlock,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.extended,
			')
	),
	'palettes' => Array (
		'1' => Array('showitem' => 't3ver_label,l18n_parent'),
//		'10' => Array('showitem' => 'fe_group'),
		'2' => Array('showitem' => 'no_auto_pb'),
		'3' => Array('showitem' => 'author_email'),
//		'4' => Array('showitem' => 'keywords'),
		'5' => Array('showitem' => 'imagealttext,--linebreak--,imagetitletext'),


	)
);



// ******************************************************************
// This is the standard TypoScript news category table, tt_news_cat
// ******************************************************************
$TCA['tt_news_cat'] = Array (
	'ctrl' => $TCA['tt_news_cat']['ctrl'],
	'interface' => Array (
		'showRecordFieldList' => 'title,image,shortcut,shortcut_target'
	),
	'columns' => Array (
		'title' => Array (
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.title',
			'config' => Array (
				'type' => 'input',
				'size' => '40',
				'max' => '256',
				'eval' => 'required'
			)
		),
		'title_lang_ol' => Array (
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.title_lang_ol',
			'config' => Array (
				'type' => 'input',
				'size' => '40',
				'max' => '256',

			)
		),
		'hidden' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.hidden',
			'config' => Array (
				'type' => 'check',
			)
		),
		'fe_group' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.fe_group',
			'config' => Array (
				'type' => 'select',
				'size' => 5,
				'maxitems' => 20,
				'items' => Array (
					Array('LLL:EXT:lang/locallang_general.php:LGL.hide_at_login', -1),
					Array('LLL:EXT:lang/locallang_general.php:LGL.any_login', -2),
					Array('LLL:EXT:lang/locallang_general.php:LGL.usergroups', '--div--')
				),
				'exclusiveKeys' => '-1,-2',
				'foreign_table' => 'fe_groups'
			)
		),
		'starttime' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.starttime',
			'config' => Array (
				'type' => 'input',
				'size' => '10',
				'max' => '20',
				'eval' => 'datetime',
				'checkbox' => '0',
				'default' => '0'
			)
		),
		'endtime' => Array (
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.php:LGL.endtime',
			'config' => Array (
				'type' => 'input',
				'size' => '8',
				'max' => '20',
				'eval' => 'datetime',
				'checkbox' => '0',
				'default' => '0',
				'range' => Array (
					'upper' => mktime(0,0,0,12,31,2020),
					'lower' => mktime(0,0,0,date('m')-1,date('d'),date('Y'))
				)
			)
		),
		'parent_category' => Array (
			'exclude' => 1,

			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.parent_category',
			'config' => Array (
				'type' => 'select',
				'type' => 'select',
				'form_type' => 'user',
				'userFunc' => 'tx_ttnews_TCAform_selectTree->renderCategoryFields',
				'treeView' => 1,
				'size' => 1,
				'minitems' => 0,
/**
 * FIXME actually maxitems is '1' but somehow the '2' is needed here
 * @see lib/class.tx_ttnews_TCAform_selectTree.php
 *
 */

				'maxitems' => 2,
				'foreign_table' => 'tt_news_cat',
//				'wizards' => Array(
//					'_PADDING' => 2,
//					'_VERTICAL' => 1,
////					'add' => Array(
////						'type' => 'script',
////						'title' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.createNewParentCategory',
////						'icon' => 'EXT:tt_news/res/add_cat.gif',
////						'params' => Array(
////							'table'=>'tt_news_cat',
////							'pid' => $sPid,
////							'setValue' => 'set'
////						),
////						'script' => 'wizard_add.php',
////					),
//					'list' => Array(
//						'type' => 'script',
//						'title' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.listCategories',
//						'icon' => 'list.gif',
//						'params' => Array(
//							'table'=>'tt_news_cat',
//							'pid' => $sPid,
//						),
//						'script' => 'wizard_list.php',
//					),
//				),

			)
		),
		'image' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.image',
			'config' => Array (
				'type' => 'group',
				'internal_type' => 'file',
				'allowed' => 'gif,png,jpeg,jpg',
				'max_size' => 100,
				'uploadfolder' => 'uploads/pics',
				'show_thumbs' => 1,
				'size' => 1,
				'minitems' => 0,
				'maxitems' => 1,
			)
		),
		'shortcut' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.shortcut',
			'config' => Array (
				'type' => 'group',
				'internal_type' => 'db',
				'allowed' => 'pages',
				'size' => '1',
				'maxitems' => '1',
				'minitems' => '0',
				'show_thumbs' => '1'
			)
		),
		'shortcut_target' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.shortcut_target',
			'config' => Array (
				'type' => 'input',
				'size' => '10',
				'checkbox' => '',
				'eval' => 'trim',
				'max' => '40'
			)
		),
		'single_pid' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.single_pid',
			'config' => Array (
				'type' => 'group',
				'internal_type' => 'db',
				'allowed' => 'pages',
				'size' => '1',
				'maxitems' => '1',
				'minitems' => '0',
				'show_thumbs' => '1'
			)
		),
		'description' => Array (
			'exclude' => 1,
			'label' => 'LLL:EXT:tt_news/locallang_tca.xml:tt_news_cat.description',
			'config' => Array (
				'type' => 'text',
				'cols' => '40',
				'rows' => '3'
			)
		),
	),

	'types' => Array (
		'0' => Array('showitem' => '
			title;;2;;1-1-1,parent_category;;;;1-1-1,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.special, image;;;;1-1-1,shortcut;;1;;1-1-1,single_pid;;;;1-1-1,description;;;;1-1-1,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.access, hidden,starttime,endtime,fe_group,
			--div--;LLL:EXT:tt_news/locallang_tca.xml:tt_news.tabs.extended,
		'),

	),
	'palettes' => Array (
		'1' => Array('showitem' => 'shortcut_target'),
		'2' => Array('showitem' => 'title_lang_ol'),
//		'10' => Array('showitem' => 'fe_group'),
	)
);


?>